adapters/audienceNetwork/facebook.go

```diff
@@ -455,18 +460,39 @@ func NewFacebookBidder(client *http.Client, platformID string, appSecret string)
 }
 
 func (fa *FacebookAdapter) MakeTimeoutNotification(req *adapters.RequestData) (*adapters.RequestData, []error) {
-	// Note, facebook creates one request per imp, so all these requests will only have one imp in them
-	auction_id, err := jsonparser.GetString(req.Body, "imp", "[0]", "id")
+	var (
+		rID   string
+		pubID string
+		err   error
+	)
+
+	// Note, the facebook adserver can only handle single impression requests, so we have to split multi-imp requests into
+	// multiple request. In order to ensure that every split request has a unique ID, the split request IDs are set to the
+	// corresponding imp's ID
+	rID, err = jsonparser.GetString(req.Body, "id")
 	if err != nil {
 		return &adapters.RequestData{}, []error{err}
 	}
 
-	uri := fmt.Sprintf("https://www.facebook.com/audiencenetwork/nurl/?partner=%s&app=%s&auction=%s&ortb_loss_code=2", fa.platformID, fa.platformID, auction_id)
+	// The publisher ID is either in the app object or the site object, depending on the supply of the request so we need
+	// to check both
+	pubID, err = jsonparser.GetString(req.Body, "app", "publisher", "id")
+	if err != nil {
+		pubID, err = jsonparser.GetString(req.Body, "site", "publisher", "id")
```

Comment:
What if the app.publisher.id exists and is empty?

Reply:
we would have rejected the impression — we validate that there is a publisher ID before sending a request to FAN (in extractPlacementAndPublisher)

Reply:
We need to consider the case where the request comes from somewhere other than Appnexus.

Reply:
The validation @Aadeshp mentions isn't specific to AppNexus, is it? The `extractPlacementAndPublisher` method is in the PBS adapter.

Reply:
Oh, yes, the validation is in the facebook adapter. So it should be safe.